profiling/src/profile/internal/mapping.rs

```diff
@@ -0,0 +1,69 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+
+use super::{Id, Item, PprofItem};
+use crate::profile::{pprof, StringId};
+use std::num::NonZeroU32;
+
+#[derive(Eq, PartialEq, Hash)]
+pub struct Mapping {
+    /// Address at which the binary (or DLL) is loaded into memory.
+    pub memory_start: u64,
+    /// The limit of the address range occupied by this mapping.
+    pub memory_limit: u64,
+    /// Offset in the binary that corresponds to the first mapped address.
+    pub file_offset: u64,
```

Comment:
Realistically, I think this can be bound to a u32.  This combined with my other comment would shave a byte off of this, but it would require a refactor at serialization.